### PR TITLE
Polymer 1.0

### PR DESCRIPTION
**Pull request by pkaske**
I tried to migrate your element to polymer ^1.0.0. I also implemented a way to change the source url for the location files.

Other changes:
Rename messageDir to messagesUrl (more fitting in my opinion).
Added possibility to set the source url for locales on the I18nMsg object.
Added observer to automatically update elements if the source url was changed.
Fixed typo in '18n-language-ready' event. Property 'language' was misspelled.
Fixed bug where new locale files wouldn't get fetched, because the 'fetching_' flag isn't cleared if a request failed before.

### CODE REVIEW

**Comment by merqurio** [comment]
Nice !

File: i18n-msg.html [hunk 1]
```diff
@@ -64,166 +65,202 @@
...
+        window.I18nMsg = window.I18nMsg || {
+            lang: null,
+            url: 'locales',
+            locales: {}
+        };
+
+        var instances = [];
+
+        var fetching_ = false;
+        /* If if there's an outstanding XHR fetching a locale file. */
```
Comment (ebidel) [inline i18n-msg.html]: Comment before the var please

File: i18n-msg.html [hunk 2]
```diff
@@ -64,166 +65,202 @@
...
+
+        window.I18nMsg = window.I18nMsg || {
+            lang: null,
+            url: 'locales',
+            locales: {}
+        };
+
+        var instances = [];
+
+        var fetching_ = false;
```
Comment (ebidel) [inline i18n-msg.html]: Change to `_fetching`. This is Polymer's new style guide for private vars

File: i18n-msg.html [hunk 3]
```diff
@@ -64,166 +65,202 @@
...
+        /* If if there's an outstanding XHR fetching a locale file. */
+
+        var numInstancesUpdated_ = 0;
```
Comment (ebidel) [inline i18n-msg.html]: same `_numInstancesUpdated`

File: i18n-msg.html [hunk 4]
```diff
@@ -64,166 +65,202 @@
...
+            },
+
+            /**
+             * The folder name where the localized `<lang>.json` files are located.
+             *
+             * @attribute messagesUrl
+             * @type string
+             * @default 'locales'
+             */
+            messagesUrl: 'locales',
```
Comment (ebidel) [inline i18n-msg.html]: put this in `properties` too and define type and default val

File: i18n-msg.html [hunk 5]
```diff
@@ -64,166 +65,202 @@
...
+                 * The set language being used.
+                 *
+                 * @property language
+                 * @type string
+                 * @default null
+                 * @readonly
+                 */
+                language: {
+                    type: String,
+                    value: null,
```
Comment (ebidel) [inline i18n-msg.html]: `readOnly: true` and you'll need to use the special method `this._setLanguage(val)` when setting it.

File: i18n-msg.html [hunk 6]
```diff
@@ -64,166 +65,202 @@
...
+             * An object containing the set of known language locales that have been loaded.
+             *
+             * @property locales
+             * @type object
+             * @default {}
+             */
+            locales: I18nMsg.locales, // static
+
+            ready: function () {
+                this.language = I18nMsg.lang;
```
Comment (ebidel) [inline i18n-msg.html]: two space indents please

File: i18n-msg.html [hunk 7]
```diff
@@ -64,166 +65,202 @@
...
+                var observerLang = new PathObserver(window.I18nMsg, 'lang');
+                observerLang.open(function (newValue, oldValue) {
+                    numInstancesUpdated_ = 0;
+                    this.language = newValue;
+                }.bind(this));
+
+                // Have instances observe window.I18nMsg.lang changes
+                // and go fetch the localized messages.json file.
+                var observerUrl = new PathObserver(window.I18nMsg, 'url');
+                observerUrl.open(function (newValue, oldValue) {
```
Comment (ebidel) [inline i18n-msg.html]: `function (` -> `function(`

File: i18n-msg.html [hunk 8]
```diff
@@ -64,166 +65,202 @@
...
+                }
+            },
+
+            languageChanged: function () {
+                // Don't fetch .json file if it has already been
+                // fetched or another instance is already trying to.
+                if (this.language && this.locales[this.language] && !fetching_) {
+                    // Send one signal that language changed to outside.
+                    if (numInstancesUpdated_ == instances.length - 1) {
+                        this.fire('i18n-language-ready', {language: this.language});
```
Comment (ebidel) [inline i18n-msg.html]: Can you document this event. SEe https://github.com/PolymerElements/seed-element/blob/master/seed-element.html#L122-L127

File: i18n-msg.html [hunk 9]
```diff
@@ -64,166 +65,202 @@
...
+
+                    this.fetchLanguage();
+                }.bind(this));
+
+                instances.push(this);
+            },
+
+            attached: function () {
+                if (!this.msgid) {
+                    this.msgid = this.innerHTML;
```
Comment (ebidel) [inline i18n-msg.html]: The intention was that if a msgid isn't found or ommitted, the `textContent` is used as fallback text. I think we should keep those semantics. Please remove.

**Comment by ebidel** [comment]
Awesome stuff! Sorry for the google io delay :)

**Comment by pkaske** [comment]
Sorry for the indentation chaos. My IDE is sometimes just to automatic...
In your comments at the beginning of the file you mention the a `getMessage` method. Was that ever implemented? I couldn't find it in the code at all.
Let me know what I can do better. I'm eager to learn from you guys!

Another question. Wouldn't it be convenient to use `iron-ajax` for fetching the locale file? To get support for request method's, cookies, auth, and so on... The ajax params for this could also be set by `window.I18nMsg`.

**Comment by ebidel** [comment]
Supe bowl indeed!

LGTM. Couple of last bits.

We could potentially move to iron-ajax in the future. I didn't see a whole lot of value using it here because most of the element is JS based and we don't need any declarative bindings in markup.

**Comment by pkaske** [comment]
Sorry, I'm not quite sure how you want `this.getMsg()` to behave. Can you clarify a little bit more. Maybe it's to late for my brain. It's midnight here in germany ;)

**Comment by pkaske** [comment]
You guys always make an awesome I/O! I hope some day I can afford to come!

**Comment by ebidel** [comment]
No worries. LGTM on the PR. I'll merge and fix up the rest. Thanks for submitting this!

**Comment by pkaske** [comment]
You're welcome! Thanks for giving the opportunity.

**Pull request merged**